kernel/src/hil/public_key_crypto/signature.rs

```diff
@@ -0,0 +1,59 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Interface for verifying signatures.
+
+use crate::ErrorCode;
+
+/// This trait provides callbacks for when the verification has completed.
+pub trait ClientVerify<const HL: usize, const SL: usize> {
+    /// Called when the verification is complete.
+    ///
+    /// If the verification operation did not encounter any errors, `result`
```

Comment:
"If the verification operation encounters an error, `result` will be a `Result::Err` specifying the `ErrorCode`. Otherwise, `result` will be a `Result::Ok` set to `Ok(true)` if the signature was correctly verified and `Ok(false)` otherwise."